dask/array/ghost.py

```diff
@@ -196,16 +196,28 @@ def nearest(x, axis, depth):
     This mimics what the skimage.filters.gaussian_filter(... mode="nearest")
     does.
     """
-    left =  ((slice(None, None, None),) * axis
-           + (slice(0, 1),)
-           + (slice(None, None, None),) * (x.ndim - axis - 1))
-    right = ((slice(None, None, None),) * axis
-           + (slice(-1, -2, -1),)
-           + (slice(None, None, None),) * (x.ndim - axis - 1))
-    l = [x[left]] * depth
-    r = [x[right]] * depth
+    left = ((slice(None, None, None),) * axis +
+            (slice(0, 1),) +
+            (slice(None, None, None),) * (x.ndim - axis - 1))
+    right = ((slice(None, None, None),) * axis +
+             (slice(-1, -2, -1),) +
+             (slice(None, None, None),) * (x.ndim - axis - 1))
+
+    l = concatenate([x[left]] * depth, axis=axis)
+    r = concatenate([x[right]] * depth, axis=axis)
+
+    # each slice in l & r is its own chunk, we want to rechunk them
+    # to have the same chunking as the array
+    lchunks = list(l.chunks)
+    lchunks[axis] = (depth,)
+    rchunks = list(r.chunks)
+    rchunks[axis] = (depth,)
+
+    l = l.rechunk(tuple(lchunks))
+    r = r.rechunk(tuple(rchunks))
```

Comment:
In `l = concatenate([x[left]] * depth)` each `x[left]` keep its own chunks. So we end up with a bunch of 1 element thick chunks along the concatenation axis. Which:
 * is really inefficient since all these get padded
 * should still work in theory but it doesn't, since the padding is larger than the chunks.

I'm working on commits on top of this to fix the padding for the large padding small chunk issues.